real estate management rest api

netabirami opened the pull request.

netabirami committed:
real estate management rest api

MenilV commented on src/main/java/com/example/RealEstateManagement/controller/PropertyController.java:
```diff
@@ -0,0 +1,29 @@
+    @Autowired
+    private PropertyService propertyService;
+
+    @GetMapping
+    public List<Property> getAllProperty(){
+        return  propertyService.getAllProperty();
+    }
+
+    @PostMapping
+    public Property addProperty (@RequestBody Property property) {
+        return propertyService.createProperty(property);
+    }
+    @DeleteMapping("/{id}")
+    public void deleteProperty(@PathVariable Long id){
+          propertyService.deleteProperty(id);
+    }
```
yes, but you're missing `GET /properties/{id}` and `PUT /properties/{id}` in this controller

netabirami replied:
yes, will update this soon

MenilV commented on src/main/java/com/example/RealEstateManagement/controller/RealEstateAgentController.java:
```diff
@@ -0,0 +1,31 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/realestateagents")
```
lets call this `/realtor`

netabirami replied:
i couldn't able to understand when it is realtor, that is what I name as realestateagents

MenilV commented on src/main/java/com/example/RealEstateManagement/model/Property.java:
```diff
@@ -0,0 +1,31 @@
+import lombok.Setter;
+
+import java.sql.Date;
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
```
nice usage of Lombok!

netabirami replied:
Thank you

MenilV reviewed (commented):
Overall great job, a few things missing, but I'll assume this is still in progress! 🚀

netabirami committed:
real estate management rest api
updated by get by ID

netabirami committed:
real estate management rest api
updated by get by ID

netabirami committed:
real estate management rest api
updated by get by ID

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller, Test case for delete and update

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller, Test case for delete and update, added SecurityConfig

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller, Test case for delete and update, added SecurityConfig, field  validation

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller, Test case for delete and update, added SecurityConfig, field  validation, Validating the building type

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller, Test case for delete and update, added SecurityConfig, field , validation, Validating the building type, validated the price max and minimum

netabirami committed:
real estate management rest api
updated by get by ID, Test Cases for controller, Test case for delete and update, added SecurityConfig, field , validation, Validating the building type, validated the price max and minimum, validated the area , bedrooms and  bathrooms
